docs/topics/ingress.md

```diff
@@ -0,0 +1,226 @@
+
+```
+$ helm install stable/nginx-ingress --name nginx-ingress
+```
+
+Once deployed, this chart will create (among other things) a Kubernetes service of type `LoadBalancer` - which should be provided by your managed Kubernetes service:
+
+```
+$ kubectl get svc
+NAME                            TYPE           CLUSTER-IP     EXTERNAL-IP    PORT(S)                      AGE
+nginx-ingress-controller        LoadBalancer   10.0.27.38     13.69.59.166   80:31875/TCP,443:31071/TCP   3m54s
+nginx-ingress-default-backend   ClusterIP      10.0.121.201   <none>         80/TCP                       3m54s
+```
+
+We will use the public IP to configure our DNS name. You can use any DNS provider - and in this example, we will use Cloudflare to add a new A Record :
+
+<img src="img/cloudflare-dns.png" style="height: 500px;" />
+
+At this point, you should be able to create Ingress objects that point to internal Kubernetes services, without exposing them. However, the ingress would not have a TLS certificate, and manually adding certificates for each one, while possible, is not ideal. This is where [`cert-manager`](https://github.com/jetstack/cert-manager),and it allows us to  _automatically provision and manage TLS certificates in Kubernetes_.
+
+## Cert-Manager
+
+Following the [official instructions on deploying `cert-manager`](https://github.com/jetstack/cert-manager/blob/master/docs/tutorials/acme/quick-start/index.rst#step-5---deploy-cert-manager):
+
+```
+# Install the cert-manager CRDs. We must do this before installing the Helm
+# chart in the next step
+$ kubectl apply -f https://raw.githubusercontent.com/jetstack/cert-manager/release-0.6/deploy/manifests/00-crds.yaml
+
+# Update your local Helm chart repositories
+$ helm repo update
+
+# Install cert-manager
+$ helm install --name cert-manager --namespace cert-manager stable/cert-manager
+```
+
+This created some Kubernetes Custom Resources Definitions (CRDs), and deployed the Helm chart. Among the CRDs created is one of type `Issuer` - when an ingress resource is annotated with a specific issuer, this will request a TLS certificate from the ACME server URL from the definition below, and before expiration, an email will be sent to the address defined below - so make sure to edit the file.
+
+Let's Encrypt defines two tiers for emitting certificates - staging and production - production certificates have very strong [limit rates](https://letsencrypt.org/docs/rate-limits/), so in order to make sure your setup works, start with generating staging certificates, then once you know it works, create a production one.
+
+Below you can find the definitions for each of the tiers:
+
+```yaml
+   apiVersion: certmanager.k8s.io/v1alpha1
+   kind: ClusterIssuer
+   metadata:
+     name: letsencrypt-staging
+   spec:
+     acme:
+       # The ACME server URL
+       server: https://acme-staging-v02.api.letsencrypt.org/directory
+       # Email address used for ACME registration
```

Comment:
you've mentioned it before - but since it's important - add a comment that this should be a *real* e-mail address so the expiration mail arrives correctly?